website/server/models/group.js

```diff
@@ -547,6 +559,8 @@ schema.methods.sendChat = function sendChat (message, user, metaData) {
 
   if (user) setUserStyles(newChatMessage, user);
 
+  checkForChatApproval(this, newMessage, user);
```

Comment:
This should be `newChatMesage` not `newMessage`. With `newMessage` the `true` value is never assigned to the `approvalRequired` field in the message. 

Also that field needs to be added to website/server/models/chat.js